src/timelist.c

```diff
@@ -90,8 +96,32 @@ void timelist_timer_start(timelist *head, const char *name)
 
 	clock_gettime(CLOCK_MONOTONIC_RAW, &t->rawstart);
 	clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &t->procstart);
-	t->name = name;
+
+	/* look for all elements in the queue and count how many of those
+	 * are still not complete, those are "parent" times */
+	TAILQ_FOREACH_REVERSE(prev_t, head, timelist, times)
+	{
+		if (prev_t->complete == false) {
+			count++;
+		}
+	}
+	/* we want to indent each one of the child times with four spaces,
+	 * starting on the second level child, we don't want any space in the
+	 * first child. */
+	if (count) {
+		number_of_spaces = (count - 1) * 4;
+	}
+	char spaces[number_of_spaces];
+	for (int i = 0; i < number_of_spaces; i++) {
+		spaces[i] = ' ';
+	}
+	spaces[number_of_spaces] = '\0';
+
+	/* build the decorated name that will be inserted */
+	string_or_die(&decorated_name, "%s%s%s", spaces, count ? "|-- " : "", name);
```

Comment:
Just a hack using "%*s". If you do this:

    string_or_die(&decorated_name, "%*s%s%s", number_of_spaces, "", count ? "|-- " : "", name);

Reply:
And you need to free memory allocated by string_or_die on timelist_free.

Reply:
> Just a hack using "%*s". If you do this:
> 
> ```
> string_or_die(&decorated_name, "%*s%s%s", number_of_spaces, "", count ? "|-- " : "", name);
> ```

Nice, I tried doing this before but it didn't work, but know I see what I was doing wrong. 😅 Thanks for the tip